### PR TITLE
add support for generating tests for functions defined in test files

### PR DESCRIPTION
**Pull request by s-kostyaev**
It can be useful if you write some wrapper function in test file to simplify testing. For example I wrote function that is not pure. And for test it I wrote some wrapper that mock this function environment and call that function multiple times on series of input data. When I was try to generate table test for this wrapper function `gotests` wan't help me :smile: So I use dirty hack to do it - move wrapper function from test file and back. And now I want better solution like in this PR.

### CODE REVIEW

**Comment by s-kostyaev** [comment]
@googlebot I signed it!

File: gotests_test.go [hunk 1]
```diff
@@ -66,10 +66,13 @@ func TestGenerateTests(t *testing.T) {
...
-				srcPath: `testdata/test100_test.go`,
+				srcPath:  `testdata/test100_test.go`,
+				only:     regexp.MustCompile("wrapToString"),
+				subtests: true,
 			},
-			wantNoTests: true,
-			wantErr:     true,
+			wantNoTests: false,
+			wantErr:     false,
+			want:        mustReadFile(t, `testdata/test100result/result.go`),
```
Comment (cweill) [inline gotests_test.go]: The target file should live under `testdata/goldens` with a name like `testdata/goldens/target_test_file.go`, as opposed to `testdata/test100result/result.go`.

File: testdata/goldens/existing_test_file.go [hunk 2]
```diff
@@ -45,6 +46,14 @@ func TestBaz100(t *testing.T) {
 	}
 }
 
+func wrapToString(in []int) []string {
```
Comment (cweill) [inline testdata/goldens/existing_test_file.go]: I think you can revert the changes in this file.

File: testdata/test100result/result.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package testdata
```
Comment (cweill) [inline testdata/test100result/result.go]: See my comment above about renaming this file and putting it in the goldens directory.

File: testdata/goldens/existing_test_file.go (hunk 2 above)
Comment (s-kostyaev) [reply testdata/goldens/existing_test_file.go]: And for which function should I generate test?

**Comment by s-kostyaev** [comment]
@cweill thanks for fast reaction

File: gotests_test.go [hunk 4]
```diff
@@ -66,10 +66,13 @@ func TestGenerateTests(t *testing.T) {
 		}, {
 			name: "Target test file",
 			args: args{
-				srcPath: `testdata/test100_test.go`,
+				srcPath:  `testdata/test103_test.go`,
+				only:     regexp.MustCompile("wrapToString"),
```
Comment (cweill) [inline gotests_test.go]: If you don't specify the "only" parameter, does it generate tests for the test functions? IRC the reason for not including "_test" files was to avoid generating tests for test functions like `TestBarBar100` becoming `TestTestBarBar100`. If so, please add a condition not to generate tests for test functions in a test file, and another unit test to verify the behavior. Otherwise this pull request looks good to me.

Comment (s-kostyaev) [reply gotests_test.go]: Ok. I will fix it.

**Comment by s-kostyaev** [comment]
Some problems with coveralls on travis:
> The command "$HOME/gopath/bin/goveralls -service=travis-ci -coverprofile=coverage.cov" exited with 1

File: gotests_test.go (hunk 4 above)
Comment (s-kostyaev) [reply gotests_test.go]: Done.

**Comment by s-kostyaev** [comment]
Thanks.
